src/index.coffee

```diff
@@ -94,8 +115,31 @@ reinstall_all = (options = {}, packages) ->
 
   return if packages then curried packages else curried
 
+shrinkwrap = (packages, options = {}) ->
+  shrinkwrap_json =
+    dependencies: {}
+
+  for pkg in packages
+    [ whole, git_at_github, name, branch] = pkg.match /^(.+?):(.+?)(?:\.git)(?:#(.+))?$/
```

Comment:
This regexp won't work for Git over HTTPS urls, e.g. `https://github.com/lzrski/npm-git-install.git`